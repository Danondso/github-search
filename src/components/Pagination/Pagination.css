.pagination {
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: wrap;
}

.page-item {
    list-style-type: none;
    border-width: 1px;
    padding: 1rem;
    font-size: medium;
}

.page-item.disabled {
    color: #BFC0C0;
}


.page-link {
    cursor: pointer;
}

.active {
    color: #A06CD5;
}