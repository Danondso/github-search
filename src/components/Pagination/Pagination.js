import React from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';

function Pagination({ handlePaginate, totalPages }) {
  return (
    <ReactPaginate
      previousLabel="Previous"
      nextLabel="Next"
      pageClassName="page-item"
      pageLinkClassName="page-link"
      previousClassName="page-item"
      previousLinkClassName="page-link"
      nextClassName="page-item"
      disabledLinkClassName="page-item-disabled"
      nextLinkClassName="page-link"
      breakLabel="..."
      breakClassName="page-item"
      breakLinkClassName="page-link"
      containerClassName="pagination"
      activeClassName="active"
      marginPagesDisplayed={2}
      pageRangeDisplayed={3}
      onPageChange={handlePaginate}
      pageCount={totalPages}
      renderOnZeroPageCount={null}
    />
  );
}

Pagination.propTypes = {
  handlePaginate: PropTypes.func.isRequired,
  totalPages: PropTypes.number.isRequired,
};

export default Pagination;
