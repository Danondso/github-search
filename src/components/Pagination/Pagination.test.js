import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Pagination from './Pagination';

const handlePaginateMock = jest.fn();
beforeEach(() => {
  handlePaginateMock.mockReset();
});

describe('Pagination Test Suite', () => {
  test('Renders with required props', () => {
    render(
      <Pagination
        handlePaginate={handlePaginateMock}
        totalPages={120}
      />,
    );
    const inputElement = screen.getByText('Next');
    expect(inputElement).toMatchSnapshot();
  });

  test('Next and Previous clicks trigger the handlePaginateFunction', () => {
    render(
      <Pagination
        handlePaginate={handlePaginateMock}
        totalPages={120}
      />,
    );
    userEvent.click(screen.getByText('Next'));
    userEvent.click(screen.getByText('Previous'));
    expect(handlePaginateMock).toHaveBeenCalledTimes(2);
  });
});
