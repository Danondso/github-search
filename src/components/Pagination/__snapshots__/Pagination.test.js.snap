// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Test Suite Renders with required props 1`] = `
<a
  aria-disabled="false"
  aria-label="Next page"
  class="page-link"
  rel="next"
  role="button"
  tabindex="0"
>
  Next
</a>
`;
